lib/Modules/JujuHelper/JujuHelper.psm1

```diff
@@ -148,7 +151,15 @@ function Invoke-FastWebRequest {
             }
         }
 
-        $client = new-object System.Net.Http.HttpClient
+        if($Proxy) {
+            $proxyObj = New-Object "System.Net.WebProxy" $Proxy
```

Comment:
This should be moved into a function of it's own.

```powershell
function Get-HTTPClientWithProxy {
    Param(
        [Parameter(Mandatory=$true)]
        [string]$ProxyURL
    )

    $proxyObj = New-Object "System.Net.WebProxy" $ProxyURL
    $handler = New-Object "System.Net.Http.HttpClientHandler"
    $handler.UseProxy = $true
    $handler.Proxy = $proxyObj
    $client = new-object System.Net.Http.HttpClient $handler

    return $client
}
```

And call it here to get an instance of a web client.

Or, I think you can use ```Invoke-FastWebRequest``` which has the ability to use a proxy.

Reply:
The code is already under Invoke-FastWebRequest, is there a reason to split it's functionality into 2 functions? (have an extra function just to create the client object: Get-HTTPClientWithProxy and call it inside Invoke-FastWebRequest)

Reply:
No. I was simply giving you the choice. ```Invoke-FastWebRequest``` is the way to go, but if you have any valid reason to use the above code, you should put it in a function.